KK_SkinEffects/Hooks.PersistClothes.cs

```diff
@@ -16,11 +17,22 @@ internal static partial class Hooks
         /// </summary>
         private static class PersistClothes
         {
-            [HarmonyPostfix]
```

Comment:
Why patch the MoveNext iterator with prefix instead of just changing this to [HarmonyPrefix]? It should have the same effect but be less brittle (generated class names can change between versions of the game and make it harder to port a plugin).

Reply:
So my understanding is that I'm patching a compiler generated private nested class, TalkScene+<TalkEnd>c__Iterator5, which handles the coroutine's iterator. I can't (don't know how to) pass that nested type to [HarmonyPrefix]. If you mean that I can still patch TalkScene.TalkEnd and get the same behavior, that would surprise me, but I'll try it out.

Reply:
Ha, just gonna resend these comments.

So my understanding is that I'm patching a compiler generated private nested class, TalkScene+c__Iterator5, which handles the coroutine's iterator. I can't (don't know how to) pass that nested type to [HarmonyPrefix]. If you mean that I can still patch TalkScene.TalkEnd and get the same behavior, that would surprise me, but I'll try it out.

Reply:
Yes, it will work essentially the same. MoveNext is called immediately after the entry method, so it doesn't matter where the prefix is. It matters only for transplier and postfix patches.

Reply:
Unless you wanted to make it into a postfix patch, in which case this line is wrong `instance.Patch(iteratorMethod, prefix, null, null);`